### DISCUSSION TITLE
Minio startup speed slow due to large amount of data

### DISCUSSION THREAD

**Discussion by jiangknight** [minio/minio#17291] [Q&A] [answered]
I am having a problem with Minio's startup speed being slow due to a large amount of data. When I start Minio, it takes a long time for it to initialize and become functional. I have tried various options to optimize the configuration and performance of Minio, but the problem remains.

I would appreciate any assistance or suggestions you have to speed up Minio's startup process.

Thank you!

**Comment by jiangknight** [comment]
To supplement, use k8s to restart mini and keep it in init state

**Comment by klauspost** [comment]
The only thing that affects minio startup speed I can think of is if you have an excessive number of buckets or if you have external users that take a long time to load.

You don't provide any useful information and generally optimizing system performance is part of what we offer in our [subsciptions](https://min.io/pricing).

**Comment by mshanmu** [comment, marked as answer]
@jiangknight , when the pod is started, the k8s changes the permission of all the files in the volumes, that can cause this slowdown. Try the below in your security context,

securityContext:
  fsGroupChangePolicy: "OnRootMismatch"

**Comment by jaykrik** [comment]
Thank you @mshanmu 
This fixed a long standing issue we were facing
